homeassistant/components/media_player/cast.py

```diff
@@ -33,90 +38,159 @@
     SUPPORT_TURN_ON | SUPPORT_TURN_OFF | SUPPORT_PREVIOUS_TRACK | \
     SUPPORT_NEXT_TRACK | SUPPORT_PLAY_MEDIA | SUPPORT_STOP | SUPPORT_PLAY
 
-KNOWN_HOSTS_KEY = 'cast_known_hosts'
+INTERNAL_DISCOVERY_RUNNING = 'cast_discovery_running'
```

Comment:
In the past, we'd just have a separate pychromecast discovery run for **every single** chromecast. That was an immense waste of resources. This PR uses only 1 discovery service at max. It needs to know though whether its own discovery service is running already, so I store it in `hass.data[cast_discovery_running]`. If this is the wrong place, please tell me.